### DISCUSSION TITLE
Loading Snomed-CT Files

### DISCUSSION THREAD

**Discussion by sschaat** [b2ihealthcare/snow-owl#1092] [Q&A] [answered]
Hi,

I am running Snowstorm in Docker. 

Now I am trying to load the Snomed Files following https://github.com/IHTSDO/snowstorm/blob/master/docs/loading-snomed.md 

I tried to create a job using the Import-Endpoint (http://snowowl_lab:8080/snowowl/?api=snomedct#post-/snomedct/-path-/import) 

- **Which path parameter should I use?**. Is it a directory that I need to create? If yes, where should I create it? I tried to create one outside and inside of the docker container. 
- For the file parameter, is it correct to upload a JSON-File with following content:
_{
  "branchPath": "MAIN",
  "codeSystemShortName": "SNOMEDCT",
  "createVersions": true,
  "type": "SNAPSHOT"
}_

Using the /jobs/{id} Endpoint (API http://snowowl_lab:8080/snowowl/?api=core#get-/jobs/-id- )
I see that the job state is "FAILED" and the result is: 

  _"result": "{\"message\":\"Resource with identifier 'snowowl' could not be found.\",\"developerMessage\":\"The requested instance resource (id = 'snowowl', type = 'Resource') does not exist and/or not yet created.\",\"code\":0,\"status\":404,\"additionalInfo\":{}}",_

Where would the job create the archive directory, so that I can upload the Snomed-Files there?

Thanks for any hint!

**Comment by cmark** [comment, marked as answer]
Hi @sschaat,

Snowstorm is a different SNOMED CT Terminology Server than Snow Owl, our documentation can be found here: https://docs.b2i.sg/snow-owl

Something is wrong with your import request URL, because it tries to recognize the `snowowl` part as a CodeSystem, which is clearly wrong.
Could you please share the actual import request `cURL` command you have used? (if you have initiated the import request from the Swagger UI then please share a screenshot of the parameters or copy the request from the `cURL` tab)

> Which path parameter should I use?. Is it a directory that I need to create? If yes, where should I create it? I tried to create one outside and inside of the docker container.

Path parameters in the Snow Owl API reference branch paths or resource URIs, not file system paths. See more info about that [here](https://docs.b2i.sg/snow-owl/index-3#path-expressions).

The usual flow to import SNOMED CT into Snow Owl is the following:

* Create the CodeSystem resource representing the SNOMED CT Edition you'd like to import. For example for the latest International Edition, one would create a Code System first (if not present already) using the CodeSystem API:

```
POST /codesystems
{
  "id": "SNOMEDCT",
  "url": "http://snomed.info/sct/900000000000207008",
  "title": "SNOMED CT International Edition",
  "language": "en",
  "description": "SNOMED CT International Edition",
  "status": "active",
  "copyright": "© 2022 International Health Terminology Standards Development Organisation 2002-2022. All rights reserved. SNOMED CT® was originally created by the College of American Pathologists. 'SNOMED' and 'SNOMED CT' are registered trademarks of International Health Terminology Standards Development Organisation, trading as SNOMED International. SNOMED CT has been created by combining SNOMED RT and a computer based nomenclature and classification known as Clinical Terms Version 3, formerly known as Read Codes Version 3, which was created on behalf of the UK Department of Health and is Crown copyright. This document forms part of the International Edition release of SNOMED CT® distributed by SNOMED International, which is subject to the SNOMED CT® Affiliate License, details of which may be found at https://www.snomed.org/snomed-ct/get-snomed.",
  "owner": "YOUR_USER_ID_HERE",
  "contact": "https://snomed.org",
  "oid": "2.16.840.1.113883.6.96",
  "toolingId": "snomed",
  "settings": {
	"moduleIds": [
		"900000000000207008",
		"900000000000012004"
	],
	"locales": [
		"en-x-900000000000508004",
		"en-x-900000000000509007"
	],
	"languages": [
		{
			"languageTag": "en",
			"languageRefSetIds": [
				"900000000000509007",
				"900000000000508004"
			]
		},
		{
			"languageTag": "en-us",
			"languageRefSetIds": [
				"900000000000509007"
			]
		},
		{
			"languageTag": "en-gb",
			"languageRefSetIds": [
				"900000000000508004"
			]
		}
	],
	"publisher": "SNOMED International",
	"namespace": "373872000"
  }
}
```

* Import the release files into the Resource's development branch (the default working branch or path):

```
POST -F file=@SnomedCT_RF2Release_INT_20221031.zip /snomedct/SNOMEDCT/import
{
  "type": "snapshot",
  "createVersions": false
}
```

Once the imported started you can monitor it with the following endpoint:

```
GET /snomedct/SNOMEDCT/import/<the job id returned in the HTTP response's Location header of the above call>
```

Or you can follow the log output of the Snow Owl service (or Docker container).

I hope this helps.

Cheers,
Mark

**Comment by sschaat** [comment]
Hi Mark, 

sorry for mixing up Snowstorm and Snowowl. That happens when you try to setup both in parallel....

I created the CodeSystem "SNOMEDCT" and was able to see it under http://snowowl_lab:8080/snowowl/codesystems
Now I used the import endpoint with the created Codesystem ("SNOMEDCT") as path-parameter

![Unbenannt](https://user-images.githubusercontent.com/113350729/204790012-74a43243-d9b8-4cc0-87ed-cfd844f086fe.PNG)

The GUI says I should upload the files via the archive endpoint, but that's probably obsolete since I used the "file" paramter in the import endpoint. I also cannot find an archive endpoint.

The job-endpoint in the CORE-API tells me the status is "RUNNING", so it seems that I have to wait...

Thanks again!

**Comment by cmark** [comment]
Hi @sschaat,

> sorry for mixing up Snowstorm and Snowowl. That happens when you try to setup both in parallel....

No problem.

> The GUI says I should upload the files via the archive endpoint, but that's probably obsolete since I used the "file" paramter in the import endpoint. I also cannot find an archive endpoint.

Yes, that is obsolete, we will get this fixed quickly.

I hope the import finishes soon and you'll have a SNOMED content loaded in Snow Owl. Let us know if it fails or if you have any further questions.

Cheers,
Mark
